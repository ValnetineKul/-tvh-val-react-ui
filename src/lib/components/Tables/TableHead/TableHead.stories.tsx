import React from 'react';
import TableCell from '../TableCell';
import TableHead from './TableHead';
import Table from '../Table';
import TableRow from '../TableRow';
import { createMeta } from '../../../story-utils';

export const Primitive = () => {
  return (
    <Table>
      <TableHead>
        <TableRow>
          <TableCell>Part</TableCell>
          <TableCell>Make</TableCell>
          <TableCell numeric>N Header</TableCell>
        </TableRow>
      </TableHead>
    </Table>
  );
};

Primitive.storyName = 'TableHead';

export default createMeta({
  component: TableHead,
  title: 'Components/Tables/TableHead',
  argTypes: {},
  parameters: {
    componentSubtitle: '',
  },
});
