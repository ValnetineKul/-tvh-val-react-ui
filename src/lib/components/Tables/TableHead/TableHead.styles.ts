import { makeStyles } from '../../../../themes/core';

const useStyles = makeStyles({ name: 'TableHead' })((theme) => ({
  root: {
    backgroundColor: theme.color['bg/surface200/default'],
  },
}));

export default useStyles;
