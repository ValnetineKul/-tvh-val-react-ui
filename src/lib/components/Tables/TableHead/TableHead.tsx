import type { FC } from 'react';
import React from 'react';
import { default as MuiTableHead } from '@mui/material/TableHead';
import useStyles from './TableHead.styles';

export interface TableHeadProps {
  className?: string;
}

const TableHead: FC<TableHeadProps> = ({ children, className }) => {
  const { classes } = useStyles();

  return (
    <MuiTableHead classes={{ root: classes.root }} className={className}>
      {children}
    </MuiTableHead>
  );
};

export default TableHead;
