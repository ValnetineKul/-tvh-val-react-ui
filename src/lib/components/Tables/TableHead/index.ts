import TableHead from './TableHead';

export type { TableHeadProps } from './TableHead';
export default TableHead;
