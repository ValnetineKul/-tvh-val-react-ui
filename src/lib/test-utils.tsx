import userEvent from '@testing-library/user-event';
import { fireEvent, render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { theme } from '../themes/tvh';
import { createTheme, ThemeProvider, SurfaceProvider } from '../themes/core';

export const muiTheme = createTheme(theme);

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <MemoryRouter>
      <ThemeProvider theme={muiTheme}>
        <SurfaceProvider color="100">{children}</SurfaceProvider>
      </ThemeProvider>
    </MemoryRouter>
  );
};

const customRender = (component: React.ReactElement, options?: Parameters<typeof render>[1]) =>
  render(component, { wrapper: AllTheProviders, ...options });

const customRenderHook: typeof renderHook = (hook, options) =>
  renderHook(hook, { wrapper: AllTheProviders, ...options });

export const initResponsiveTest = (size: number | 'Mobile' | 'Tablet' | 'Desktop' | 'LgDesktop' | 'XlDesktop') => {
  let width: number;
  if (size === 'Mobile') width = muiTheme.breakpoints.values.sm - 1;
  else if (size === 'Tablet') width = muiTheme.breakpoints.values.sm;
  else if (size === 'Desktop') width = muiTheme.breakpoints.values.md;
  else if (size === 'LgDesktop') width = muiTheme.breakpoints.values.lg;
  else if (size === 'XlDesktop') width = muiTheme.breakpoints.values.xl;
  else width = size;

  Object.defineProperty(window, 'innerWidth', {
    writable: true,
    configurable: true,
    value: width,
  });

  window.matchMedia = jest.fn().mockImplementation((query) => {
    const queryWidthInterval: string[] = query.match(/\d+(\.\d+)?/g);

    let matches;
    if (queryWidthInterval.length === 2) {
      matches = width >= +queryWidthInterval[0] && width < +queryWidthInterval[1];
    } else {
      const queryWidth = +queryWidthInterval[0];
      const isDown = /max/.test(query);

      matches = isDown ? width <= queryWidth : width >= queryWidth;
    }

    return {
      matches,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
    };
  });
};

const customUserEvent = Object.assign(userEvent, {
  mouseDown: fireEvent.mouseDown,
  mouseUp: fireEvent.mouseUp,
  mouseMove: fireEvent.mouseMove,
  scroll: fireEvent.scroll,
  error: fireEvent.error,
});

export { screen, within, waitFor, act, waitForElementToBeRemoved } from '@testing-library/react';
// override render method
export { customRender as render, customRenderHook as renderHook, customUserEvent as userEvent };
