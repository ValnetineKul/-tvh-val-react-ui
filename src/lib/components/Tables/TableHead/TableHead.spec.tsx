import React from 'react';
import { render } from '../../../test-utils';
import TableHead from './TableHead';

describe('TableHeader', () => {
  it('Should render correctly', () => {
    const { container } = render(
      <table>
        <TableHead>
          <tr>
            <td>Children</td>
          </tr>
        </TableHead>
      </table>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
