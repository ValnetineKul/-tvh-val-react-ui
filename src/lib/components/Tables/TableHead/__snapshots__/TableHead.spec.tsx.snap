// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader Should render correctly 1`] = `
<table>
  <thead
    class="tss-1q9inhp-TableHead-root MuiTableHead-root mui-15wwp11-MuiTableHead-root"
  >
    <tr>
      <td>
        Children
      </td>
    </tr>
  </thead>
</table>
`;
